tests/unit/dense_conjugate.cpp

```diff
@@ -0,0 +1,194 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <vector>
+#include <memory>
+#include <complex>
+
+#include <alp.hpp>
+#include "../utils/print_alp_containers.hpp"
+
+constexpr float tol = 1.e-10;
+
+template< typename T >
+T random_value();
+
+template<>
+float random_value< float >() {
+	return static_cast< float >( rand() ) / RAND_MAX;
+}
+
+template<>
+std::complex< float > random_value< std::complex< float > >() {
```

Comment:
I would set alias for `folat` at one place (beginning of this file) so it can be easily changed